### PR TITLE
METRON-716 site-book README.md

### PR DESCRIPTION
**Pull request by ottobackwards**
Initial draft of a readme for site-book documentation.

Questions:  should we list the tools used more explicitly?

### CODE REVIEW

File: site-book/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Site Book Metron documentation 
+
+Metron's Site Book is an attempt at producing documentation that is
+
+-Versioned and reviewed,
```
Comment (JonZeolla) [inline site-book/README.md]: Probably want to put a space between -V and remove the comma at the end to get:
\- Versioned and reviewed

File: site-book/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+Currently it is in a stand-alone, versioned-with-the-code sub-directory and sub-project. The idea is that a release
+manager would build the site-book (following the instructions below), then copy it into a versioned subdirectory of the (unversioned) public site, to publish it along with each code release.
+
+To build the book, do the following:
+In any git clone of incubator-metron containing the site-book subdirectory,
+
+    cd site-book
+    bin/generate-md.sh
+    mvn site:site
```
Comment (JonZeolla) [inline site-book/README.md]: Perhaps should wrap in ``` for consistency?  I think that's how we do it in our other READMEs.

File: site-book/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+It only takes a few seconds. You may now view your copy of the book in a browser by opening 
+
+    file:///your/path/to/incubator-metron/site-book/target/site/index.html.
```
Comment (JonZeolla) [inline site-book/README.md]: Remove the trailing .

File: site-book/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    bin/generate-md.sh
+    mvn site:site
+
+It only takes a few seconds. You may now view your copy of the book in a browser by opening 
+
+    file:///your/path/to/incubator-metron/site-book/target/site/index.html. 
+
+On a Mac, you can just type the following on the command line
+
+    open target/site/index.html
```
Comment (JonZeolla) [inline site-book/README.md]: Wrap in ``` ?

File: site-book/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+The key files under site-book/ are:
+
+###bin/generate-md.sh
+
+- copies all .md files from the code directory tree into the site tree, performs some transformations on them, and generates the nav tree structure and labels.
```
Comment (JonZeolla) [inline site-book/README.md]: I suggest changing this to:
- Copies all .md files from the code directory tree into the site tree
- Performs some transformations on them
- Generates the nav tree structure and labels

File: site-book/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+###bin/fix-md-dialect.awk
+
+- is called by 'generate-md.sh'. It does transforms within the text of each file, related to converting the Github-MD dialect of markdown into the doxia-markdown dialect.
```
Comment (JonZeolla) [inline site-book/README.md]: I suggest changing this to:
- Called by `generate-md.sh`
- Performs transforms within the text of each file into the doxia-markdown dialect (related to converting the Github-MD dialect of markdown).

File: site-book/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+- Tied to code versions
+- Highly local to the code being documented
+
+Currently it is in a stand-alone, versioned-with-the-code sub-directory and sub-project. The idea is that a release
+manager would build the site-book (following the instructions below), then copy it into a versioned subdirectory of the (unversioned) public site, to publish it along with each code release.
+
+To build the book, do the following:
+In any git clone of incubator-metron containing the site-book subdirectory,
+
+```cd site-book
```
Comment (mattf-horton) [inline site-book/README.md]: need a newline after the triple-backtick.

File: site-book/README.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
+# Site Book Metron documentation 
+
+Metron's Site Book is an attempt at producing documentation that is
+
+- Versioned and reviewed,
+- Only committers can edit
```
Comment (mattf-horton) [inline site-book/README.md]: I would recommend deleting "only committers can edit".  It is actually self-evident, and seems unnecessary to state as a primary objective.  The other three are tremendously more important.

File: site-book/README.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
+# Site Book Metron documentation
```
Comment (mattf-horton) [inline site-book/README.md]: Suggest "Metron Site-Book documentation"

File: site-book/README.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+Metron's Site Book is an attempt at producing documentation that is
+
+- Versioned and reviewed,
+- Only committers can edit
+- Tied to code versions
+- Highly local to the code being documented
+
+Currently it is in a stand-alone, versioned-with-the-code sub-directory and sub-project. The idea is that a release
+manager would build the site-book (following the instructions below), then copy it into a versioned subdirectory of the (unversioned) public site, to publish it along with each code release.
```
Comment (mattf-horton) [inline site-book/README.md]: suggest instead this sentence to read:
"The idea is that a release manager would build the site-book (following the instructions below), then publish it from the public Metron site (http://metron.incubator.apache.org/) as the docs for the new released version.  Older site-book versions should remain available for users that need them."

File: site-book/README.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+The key files under site-book are:
+
+###bin/generate-md.sh
+
+- Copies all .md files from the code directory tree into the site tree
+- Performs some transformations on them
+- Generates the nav tree structure and labels
+
+###bin/fix-md-dialect.awk
```
Comment (JonZeolla) [inline site-book/README.md]: This was migrated to python and is now `bin/fix-md-dialect.py`.

File: site-book/README.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+- Is called by 'generate-md.sh'
+- It does transforms within the text of each file
+    - Converts the Github-MD dialect of markdown into the doxia-markdown dialect
+
+###pom.xml and src/site/site.xml
+
+- doxia boilerplate, tweaked for our specific needs.
```
Comment (JonZeolla) [inline site-book/README.md]: Doxia should be capitalized, maybe also link to https://maven.apache.org/doxia/

File: site-book/README.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
+# Metron Site-Book documentation
+
+Metron's Site Book is an attempt at producing documentation that is
```
Comment (JonZeolla) [inline site-book/README.md]: Maybe I'm being a bit of a pedant, but I would put a \: at the end of this line.

File: site-book/README.md [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+
+- Versioned and reviewed
+- Tied to code versions
+- Highly local to the code being documented
+
+The idea is that a release manager would build the site-book (following the instructions below), then publish it from the public [Metron site](http://metron.incubator.apache.org/) as the docs for the new released version. Older site-book versions should remain available for users that need them.
```
Comment (JonZeolla) [inline site-book/README.md]: Do we actually have a process for the release manager to follow that includes this?  It should probably exist [here](https://cwiki.apache.org/confluence/display/METRON/Release+Process).  If I'm not missing it, I will happily suggest an update to the wiki via the dev list.

File: site-book/README.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+
+To build the book, do the following:
```
Comment (JonZeolla) [inline site-book/README.md]: Add two spaces to the end of this line to ensure the next line says separate post-parsing.  I think you can also just add second newline between them.

File: site-book/README.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+file:///your/path/to/incubator-metron/site-book/target/site/index.html
+```
+
+On a Mac, you can just type the following on the command line
+
+```
+open target/site/index.html
+```
+
+The key files under site-book are:
```
Comment (JonZeolla) [inline site-book/README.md]: I would suggest changing this into

## Key components

However, either way, I'm fine with it.

File: site-book/README.md [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+###bin/generate-md.sh
+
+- Copies all .md files from the code directory tree into the site tree
+- Performs some transformations on them
+- Generates the nav tree structure and labels
+
+###bin/fix-md-dialect.awk
+
+- Is called by 'generate-md.sh'
+- It does transforms within the text of each file
```
Comment (JonZeolla) [inline site-book/README.md]: Suggest changing this into:
- Called by `generate-md.sh`
- Does transforms within the text of each file

File: site-book/README.md (hunk 14 above)
Comment (mattf-horton) [inline site-book/README.md]: Please do.

File: site-book/README.md [hunk 18]
```diff
@@ -1,13 +1,14 @@
...
+Metron's Site Book is an attempt at producing documentation that is:
 
 - Versioned and reviewed
 - Tied to code versions
 - Highly local to the code being documented
 
 The idea is that a release manager would build the site-book (following the instructions below), then publish it from the public [Metron site](http://metron.incubator.apache.org/) as the docs for the new released version. Older site-book versions should remain available for users that need them.
 
+
 To build the book, do the following:
```
Comment (JonZeolla) [inline site-book/README.md]: I meant adding an additional newline at the end of this line.  If you [view this line](https://github.com/ottobackwards/incubator-metron/blob/31b517aeea1e4afb73e45910db523c2660aa8584/site-book/README.md) post-parsing you'll see lines 12 and 13 are both together right now.

**Comment by JonZeolla** [comment]
+1 (non-binding) by inspection.  Thanks for putting this together Otto, good catch.

On a related note, it looks like Travis is either backed up or having an issue.  Another PR which is a couple of hours old also hasn't started yet.

File: site-book/README.md (hunk 14 above)
Comment (JonZeolla) [inline site-book/README.md]: Done.

**Comment by ottobackwards** [comment]
@mattf-horton are you ok on this after the revisions?

**Comment by mattf-horton** [comment]
+1 More than okay!  This is great, thanks for doing it.

**Pull request closed without merging**